src/coreclr/jit/valuenum.cpp

```diff
@@ -2675,7 +2675,7 @@ ValueNum ValueNumStore::EvalFuncForConstantArgs(var_types typ, VNFunc func, Valu
             {
                 assert((typ == TYP_BYREF) || (typ == TYP_LONG));
                 size_t resultVal = EvalOp<size_t>(func, arg0Val, arg1Val);
-                result           = VNForByrefCon(resultVal);
+                result           = VNForByrefCon((target_size_t)resultVal);
```

Comment:
Nit: Use `static_cast<>` here and below to conform with https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#1511-casts

Reply:
Shouldn't we update types of `arg0Val` and `arg1Val1` to be `target_size_t`?

Reply:
Currently we aren't using static_cast<> in the valuenum.cpp.
If this is desirable then we can open a cleanup issue to change to use static_cast's

Reply:
I would rather not change the types of arg0Val and argt1Val here